controllers/datadogagent/service.go

```diff
@@ -146,6 +148,33 @@ func (r *Reconciler) manageAdmissionControllerService(logger logr.Logger, dda *d
 	return r.updateIfNeededAdmissionControllerService(logger, dda, service)
 }
 
+var testGitVersion string
+
+func (r *Reconciler) manageAgentService(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
+	// Service Internal Traffic Policy exists in Kube 1.21 but it is enabled by default since 1.22
+	gitVersion := version.Get().GitVersion
+	if testGitVersion != "" {
+		gitVersion = testGitVersion
+	}
+
+	forceLocalServiceEnable := dda.Spec.Agent.LocalService != nil && dda.Spec.Agent.LocalService.ForceLocalServiceEnable != nil && *dda.Spec.Agent.LocalService.ForceLocalServiceEnable
+	if !utils.IsAboveMinVersion(gitVersion, "1.22-0") && !forceLocalServiceEnable {
```

Comment:
should it be '1.22.0" instead of "1.22-0" ?

Reply:
It was `1.22-0` on purpose.
`1.22` is equivalent to `1.22.0`.
The `-0` suffix is meant to match pre-release version. See [doc here](https://github.com/Masterminds/semver#working-with-prerelease-versions).
Concretely, whereas `1.23.1` matches both `>= 1.22` and `>= 1.22.0`, `1.23.1-rc.1` wouldn’t match any of those two.
But `1.23.1-rc.1` matches both `>= 1.22-0` and `>= 1.22.0-0` (which are equivalent).